enterprise/server/ociregistry/ociregistry.go

```diff
@@ -460,6 +471,18 @@ func fetchBlobOrManifestFromCache(ctx context.Context, w http.ResponseWriter, bs
 }
 
 func writeBlobOrManifestToCacheAndResponse(ctx context.Context, upstream io.Reader, w io.Writer, bsClient bspb.ByteStreamClient, acClient repb.ActionCacheClient, ref gcrname.Reference, ociResourceType ocipb.OCIResourceType, hash gcr.Hash, contentType string, contentLength int64) error {
+	r := upstream
+	if ociResourceType == ocipb.OCIResourceType_MANIFEST {
```

Comment:
nit: maybe clearer to move this new logic into the caller, since this function is already pretty complex

Reply:
I hear you, but the caller is already gigantic.